src/target/riscv/riscv.c

```diff
@@ -1525,8 +1532,86 @@ static int riscv_checksum_memory(struct target *target,
 		target_addr_t address, uint32_t count,
 		uint32_t *checksum)
 {
-	*checksum = 0xFFFFFFFF;
-	return ERROR_TARGET_RESOURCE_NOT_AVAILABLE;
```

Comment:
So in the old version of this function that returns resource not available, OpenOCD would perform a slower checksum operation without using a target program?

Reply:
That's right. If this function returns any kind of error, then OpenOCD falls back to reading the memory from the target, and computing the CRC itself.